src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationNameTelemetryInitializer.cs

```diff
@@ -15,7 +15,7 @@
 
     public class OperationNameTelemetryInitializer : TelemetryInitializerBase
     {
-        public const string BeforeActionNotificationName = "Microsoft.AspNet.Mvc.BeforeAction";
+        public const string BeforeActionNotificationName = "Microsoft.AspNetCore.Mvc.BeforeAction";
```

Comment:
Got the latest bits (`rc2-20634`), and updated MVC functional test accordingly, and seems like it is picking the onbeforeaction now. I am not sure where the issue is, but everything seem to run fine. 

Just to confirm, AppInsights now uses `*` as opposed to a specific version. As a result, the test application having a fixed older version throws an `typeload exception` on running. However, updating the test application to pick the latest bits solved the issue.